ogr/ogrfielddefn.cpp

```diff
@@ -108,6 +108,63 @@ OGRFieldDefn::~OGRFieldDefn()
     CPLFree(pszDefault);
 }
 
+/************************************************************************/
+/*                             OGRFieldDefn::OGRFieldDefn()             */
+/************************************************************************/
+
+/**
+ * @brief OGRFieldDefn::OGRFieldDefn copy constructor.
+ * @param oOther the object to copy.
+ * @since GDAL 3.11
+ */
+OGRFieldDefn::OGRFieldDefn(const OGRFieldDefn &oOther)
+    : pszName(CPLStrdup(oOther.pszName)),
+      pszAlternativeName(CPLStrdup(oOther.pszAlternativeName)),
+      eType(oOther.eType), eJustify(oOther.eJustify), nWidth(oOther.nWidth),
+      nPrecision(oOther.nPrecision), pszDefault(CPLStrdup(oOther.pszDefault)),
+      bIgnore(oOther.bIgnore), eSubType(oOther.eSubType),
+      bNullable(oOther.bNullable), bUnique(oOther.bUnique),
+      m_osDomainName(oOther.m_osDomainName), m_osComment(oOther.m_osComment),
+      m_nTZFlag(oOther.m_nTZFlag), m_bSealed(oOther.m_bSealed)
+{
+}
+
+/************************************************************************/
+/*                           OGRFieldDefn::operator=()                  */
+/************************************************************************/
+
+/**
+ * @brief OGRFieldDefn::operator = assignment operator.
+ * @param oOther the object to copy.
+ * @return the current object.
+ * @since GDAL 3.11
+ */
+OGRFieldDefn &OGRFieldDefn::operator=(const OGRFieldDefn &oOther)
+{
+    if (&oOther != this)
+    {
+        CPLFree(pszName);
+        pszName = CPLStrdup(oOther.pszName);
+        CPLFree(pszAlternativeName);
+        pszAlternativeName = CPLStrdup(oOther.pszAlternativeName);
+        eType = oOther.eType;
+        eJustify = oOther.eJustify;
+        nWidth = oOther.nWidth;
+        nPrecision = oOther.nPrecision;
+        CPLFree(pszDefault);
+        pszDefault = CPLStrdup(oOther.pszDefault);
```

Comment:
Given that CPLStrdup(nullptr) return an empty string and not a nullptr:

```suggestion
        pszDefault = oOther.pszDefault ? CPLStrdup(oOther.pszDefault) : nullptr;
```

Reply:
done